src/Scene/Scene.js

```diff
@@ -138,18 +130,54 @@ Scene.prototype.scheduleUpdate = function scheduleUpdate(forceRedraw) {
 };
 
 Scene.prototype.update = function update() {
-    for (var l = 0; l < this.layers.length; l++) {
-        var layer = this.layers[l].node;
-
-        for (var sl = 0; sl < layer.children.length; sl++) {
-            var sLayer = layer.children[sl];
-
-            if (sLayer instanceof Quadtree) {
-                this.browserScene.updateQuadtree(this.layers[l], this.map.layersConfiguration, this.currentCamera());
-            } else if (sLayer instanceof MobileMappingLayer) {
-                this.browserScene.updateMobileMappingLayer(sLayer, this.currentCamera());
-            } else if (sLayer instanceof Layer) {
-                this.browserScene.updateLayer(sLayer, this.currentCamera());
+    this.camera.updateMatrixWorld();
+
+    // Browse Layer tree
+    const config = this.layersConfiguration;
+
+    // parent-free layers (hack for now)
+    const parentFreeLayers = config.getGeometryLayers().filter(l => !l.graftedOnId);
+
+    // TODO?
+    const context = {
+        camera: this.camera,
+        scheduler: this.scheduler,
+        scene: this,
+    };
+
+    for (const layer of parentFreeLayers) {
+        let graftedLayers =
+            config.getColorLayers().filter(l => l.graftedOnId === layer.id);
+        graftedLayers = graftedLayers.concat(config.getElevationLayers().filter(l => l.graftedOnId === layer.id));
+        graftedLayers = graftedLayers.concat(config.getGeometryLayers().filter(l => l.graftedOnId === layer.id));
+
+        if (layer.preUpdate) {
+            layer.preUpdate(context, layer);
+        }
+
+        const elements = [];
+        // init update, with a element-less call to update
+        layer.update(context, layer).forEach(e => elements.push(e));
+
+
+        // TODO: or we could simply iterate on children, using the appropriate
+        // layer update (aka: node.layer)?
+
+        // then update for real
+        for (let i = 0; i < elements.length; i++) {
```

Comment:
Seems generic to me actually: at each step the layer returns the elements he wants to get updated 
It's similar to BrowseTree that kept updating if the process() function returned true but without forcing to use a parent/children structure.

But if you have something simpler I'm all ears :)